src/components/calcite-input-time-picker/calcite-input-time-picker.tsx

```diff
@@ -0,0 +1,330 @@
+    if (hasNumbers && !hasLetters) {
+      return true;
+    }
+    return false;
+  }
+
+  private setInputEl = (el: HTMLCalciteInputElement): void => {
+    this.inputEl = el;
+  };
+
+  private parseTimeString = (value: string): string => {
+    if (value) {
+      const splitValue = value.split(":");
+      if (splitValue.length > 1) {
+        const hour = splitValue[0];
+        const minute = splitValue[1];
+        const second = splitValue[2];
+        const hourAsNumber = parseInt(splitValue[0]);
+        const minuteAsNumber = parseInt(splitValue[1]);
+        const secondAsNumber = parseInt(splitValue[2]);
+        const hourValid =
+          hour &&
+          this.stringContainsOnlyNumbers(hour) &&
+          !isNaN(hourAsNumber) &&
+          hourAsNumber >= 0 &&
+          hourAsNumber < 24;
+        const minuteValid =
+          minute &&
+          this.stringContainsOnlyNumbers(minute) &&
+          !isNaN(minuteAsNumber) &&
+          minuteAsNumber >= 0 &&
+          minuteAsNumber < 60;
+        const secondValid =
+          second &&
+          this.stringContainsOnlyNumbers(second) &&
+          !isNaN(secondAsNumber) &&
+          secondAsNumber >= 0 &&
+          secondAsNumber < 60;
+        if ((hourValid && minuteValid && !second) || (hourValid && minuteValid && secondValid)) {
+          let newValue = `${formatNumberAsTimeString(hourAsNumber)}:${formatNumberAsTimeString(
+            minuteAsNumber
+          )}`;
+          if (secondValid && this.step !== 60) {
+            newValue = `${newValue}:${formatNumberAsTimeString(secondAsNumber)}`;
+          }
+          return newValue;
+        }
+      }
+    }
+    return null;
+  };
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  connectedCallback() {
+    this.guid = this.el.id || `calcite-input-time-picker-${guid()}`;
+  }
+
+  componentDidLoad() {
+    if (this.value) {
+      this.inputEl.value = this.parseTimeString(this.value);
+    }
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  render(): VNode {
+    const { hour, minute, second } = this.convertStringToTime(this.value);
+    return (
+      <Host>
+        <calcite-input
+          disabled={this.disabled}
+          icon="clock"
+          id={`${this.guid}-calcite-input`}
+          name={this.name}
+          onCalciteInputBlur={this.inputBlurHandler}
+          onCalciteInputFocus={this.inputFocusHandler}
+          onCalciteInputInput={this.inputInputHandler}
+          ref={this.setInputEl}
+          scale={this.scale}
+          step={this.step}
+          theme={this.theme}
+          type="time"
+          value={this.value}
+        />
+        <calcite-popover
```

Comment:
I'm not sure we want to use a popover here. The other components like dropdown/datepicker implement popper.js instead so that they can control some styling as well as aria roles etc.

Reply:
I initially went down this road but it felt overly complicated and overkill when I could just use the `popover` component directly.  Can we expose styling and aria attributes on `calcite-popover` instead?  I think this is a great usage pattern and it saved me a ton of time compared to implementing popperjs directly in the component.  I also got @bstifle to sign off on the design of the popover including the arrow in the popup "bubble" that appears.

Reply:
Also, the only styling need I had was to make the corners of the popover rounded, so I added a `corner-appearance` attribute to calcite-popover.  

What aria roles would we need to set on the popover specifically for this component?

Reply:
I think for this component, we would want to use activedescendant and not have the popover be focused. 

Can we just make the implementing popper.js easier?

Reply:
I like the idea of reusing the popover for consistency and ease, but I may be missing something.

> Can we just make the implementing popper.js easier?

You mean by adding more utilities or a functional/internal component or how so?

Reply:
@jcfranco I'm not sure. I was wondering what is complex about implementing it now.

> I like the idea of reusing the popover for consistency and ease, but I may be missing something.

The issue is that popover uses a role of dialog and tooltip uses a role of tooltip. However, these menus should probably use a listbox role.

The popover also has styling embedded with how a popover works/looks. I'm not sure that this case would work with that styling. All the other form elements with menus (datepicker, combobox, dropdown) have styling that is different than a popover.

Reply:
Updated `input-time-picker` to be a `combobox`, `time-picker` to be `listbox` and left popover as a `dialog`.  Also added `group` roles to the divs that wrap each number input with its corresponding up and down buttons: https://github.com/Esri/calcite-components/pull/1736/commits/1963c98668f9a085ed1375a74383c26a7f3c7ba6

Reply:
More recent aria fixes: https://github.com/Esri/calcite-components/pull/1736/commits/017cfb8279df38e059f31e2486f32255f650a819